### PR TITLE
Cu 1kc8162 score typescript interface

### PR DESCRIPTION
**Pull request by bzierk**
## Checklist

- [ ] UAT
- [ ] Passes final QA checks

### CODE REVIEW

**Comment by bravetarget** [comment]
Task linked: [CU-1kc8162 FACTORY - SCORE: Deposit ship to escrow](https://app.clickup.com/t/1kc8162)

File: src/util/scoreHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+ * @param payer 
+ * @param owner 
+ * @param mint - Asset mint to create token account for
+ */
+export async function createAssociatedTokenAccountInstruction(
+  payer: web3.PublicKey,
+  owner: web3.PublicKey,
+  mint: web3.PublicKey
+): Promise<web3.TransactionInstruction> {
+  const [associatedTokenAccount] = await getAtaForMint(mint, payer);
```
Comment (turnyay) [inline src/util/scoreHelpers.ts:69]: payer should be owner here

File: src/util/scoreHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+ * @param payer (Optional - if not provided, defaults to provider wallet)
+ * @param owner (Optional - if not provided, defaults to provider wallet)
+ */
+export async function createATokenAccount(
+  provider: Provider,
+  mint: web3.PublicKey,
+  payer?: web3.PublicKey,
+  owner?: web3.PublicKey
+): Promise<web3.PublicKey> {
+  const [aTokenAccount] = await getAtaForMint(mint, provider.wallet.publicKey);
```
Comment (turnyay) [inline src/util/scoreHelpers.ts:41]: move this below the owner check and set the pubkey to owner

File: src/util/scoreHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+): Promise<web3.TransactionSignature> {
+  const tx = new web3.Transaction();
+
+  mintAuthority = mintAuthority || provider.wallet.publicKey
+
+  tx.add(Token.createMintToInstruction(
+    TOKEN_PROGRAM_ID,
+    mint,
+    associatedTokenAccount,
+    provider.wallet.publicKey,
```
Comment (turnyay) [inline src/util/scoreHelpers.ts:143]: this should be mintAuthority right?

File: src/util/scoreHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+ * @param provider
+ * @param mint - Asset mint
+ * @param associatedTokenAccount - Account for minted tokens to be deposited into
+ * @param amount - Desired number of tokens to be minted
+ * @param mintAuthority - Publickey of mint authority
+ */
+export async function mintTokens(
+  provider: Provider,
+  mint: web3.PublicKey,
+  associatedTokenAccount: web3.PublicKey,
```
Comment (turnyay) [inline src/util/scoreHelpers.ts:131]: rename this to just tokenAccount ?

File: src/util/scoreHelpers.ts [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    freezeAuthority,
+  );
+  tx.add(initialInstruction);
+
+  const txid = await provider.send(tx, [account]);
+  console.log('Created mint: ', txid);
+
+  return account.publicKey
+
+
```
Comment (turnyay) [inline src/util/scoreHelpers.ts:195]: delete empty lines

File: src/util/scoreHelpers.ts [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+ */
+export async function confirmTokenBalance(
+  provider: Provider,
+  tokenAccount: web3.PublicKey,
+  expectedQuantity: number
+) {
+  const tokenData = await provider.connection.getAccountInfo(tokenAccount, 'recent');
+  const tokenAmount = byteArrayToLong(tokenData.data.slice(64, 72));
+
+  assert(tokenAmount == expectedQuantity);
```
Comment (turnyay) [inline src/util/scoreHelpers.ts:213]: you can add a message to help debug
something like "On-chain Token amount of <> does not match expected amount <>"

**Review by turnyay — CHANGES_REQUESTED** [review]

**Review by turnyay — APPROVED** [review]

**Pull request merged**